### ISSUE TITLE
Highlight other occurrences of an identifier when selected

### ISSUE THREAD

**Issue by sprt** [joefitzgerald/go-plus#413]
Suggestion: selecting `file` shouldn't highlight `filename`.

Also, it would be nice if it were smart about it:

``` go
func Foo(f string) { return f }
filename := "foo.go"
Foo(filename)
```

Selecting `filename` here would highlight `f` as well.  But I guess we need proper tools for that.

:sparkles:

**Comment by zmb3** [comment]
vim-go accomplishes this with `guru`:

Run a command similar to `guru -json what file.go:#123` and look at the `sameids` array in the JSON output.

**Comment by joefitzgerald** [comment]
Fixed in #583.
